lib/tasks/benefit_application_terminated_list.rake

```diff
@@ -0,0 +1,56 @@
+require 'csv'
+namespace :reports do
+  namespace :shop do
+    # Following task will generate daily report with terminated benefit applications
+    # RAILS_ENV=production bundle exec rake reports:shop:benefit_application_terminated_list['termination_date']
+    # RAILS_ENV=production bundle exec rake reports:shop:benefit_application_terminated_list[02/01/2017']
+
+    desc "Report of Benefit Applications Terminated"
+    task :benefit_application_terminated_list, [:termination_date] => :environment do |task, args|
+      include Config::AcaHelper
+
+      window_date = Date.strptime(args[:termination_date], "%m/%d/%Y")
+
+      valid_states = [:termination, :termination_pending]
+      terminated_sponsorships = BenefitSponsors::BenefitSponsorships::BenefitSponsorship.where({
+        "benefit_applications" => {
+          "$elemMatch" => {
+            "terminated_on" => window_date,
```

Comment:
Based on the valid states, the query would pull the same application twice, if its a future termination. One - termination_pending, two - terminated.